crates/matrix-sdk-crypto/src/session_manager/group_sessions/share_strategy.rs

```diff
@@ -225,6 +241,41 @@ fn split_recipients_withhelds_for_user(
     RecipientDevices { allowed_devices: recipients, denied_devices_with_code: withheld_recipients }
 }
 
+fn split_recipients_withhelds_for_user_based_on_identity(
```

Comment:
This function duplicates a lot of code from the other split function, though since we're ignoring blacklisted devices it isn't quite easy to not make a mess out of a function that merges both strategies.

I guess that's fine.